src/lsm/compaction.zig

```diff
@@ -410,10 +411,12 @@ pub fn CompactionType(
             // `index_block` is only valid for this callback, so copy its contents.
             // TODO(jamii) This copy can be avoided if we bypass the cache.
             stdx.copy_disjoint(.exact, u8, compaction.index_block_a, index_block);
+
+            const index_a = TableIndexSchema.from_index_block(compaction.index_block_a);
```

Comment:
This is great. 🚀 

```suggestion
            const schema_index_block_a = SchemaIndexBlock.from(compaction.index_block_a);
```

* Clarify that `schema_index_block_a` is a schema and not an index block.
* Shorten the `from` method name since the type name is clear.
* Use MSB order to put Schema first.

Reply:
It's great to clarify that it's a schema, not an index block. However, I think `Schema` may not be the MSB, e.g `filter_block` vs `filter_bytes` vs `filter_schema` and `data_block` vs `data_schema`.